contrib/cloud.google.com/go/pubsub.v1/option.go

```diff
@@ -5,9 +5,25 @@
 
 package pubsub
 
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/namingschema"
+)
+
 type config struct {
-	serviceName string
-	measured    bool
+	serviceName     string
+	publishSpanName string
+	receiveSpanName string
+	measured        bool
+}
+
+func defaults(cfg *config) {
+	cfg.serviceName = namingschema.NewServiceNameSchema(
+		"",
+		"",
+		namingschema.WithVersionOverride(namingschema.SchemaV0, ""),
+	).GetName()
+	cfg.publishSpanName = namingschema.NewGCPPubsubOutboundOp().GetName()
+	cfg.receiveSpanName = namingschema.NewGCPPubsubInboundOp().GetName()
```

Comment:
This function won't set all fields, so it won't always result in the same config, which feels wrong for a `defaults` function.

It would be better to have something like
```
func defaults() *config {
    return &config {
        ...
    }
}
```

Reply:
I agree I prefer this pattern, however I was just trying to be consistent with the rest of the contrib packages that are following this one ([ref](https://github.com/DataDog/dd-trace-go/search?q=defaults%28)).

Please let me know if the consistency issue is not a concern so I can change it 👍

Reply:
Hmm, that's unfortunate. I didn't realize how common that pattern is. 

I'm not concerned about maintaining this consistency since it's entirely internal, but I'm also ok leaving it as-is seeing how prevalent the pattern is. If you prefer my suggestion, go for it.

Maybe we can refactor the rest some day.

Reply:
changed and renamed to `defaultConfig`